### PR TITLE
Add support for I2C on STM32F7 family and on STM32F723E-DISCO board

### PR DESCRIPTION
**Pull request by aurel32**
This patch series adds support for I2C bus of the STM32F7 family, enables it on the STM32F723E-DISCO board, and finally fixes an issue found while testing the I2C support in interrupt mode.

I have tested it using by communicating with the WM8994 chip on the STM32F723E-DISCO board.

### CODE REVIEW

**Review by ydamigos — CHANGES_REQUESTED** [review]
Looks good. I am just blocking it because PR #7626 also introduces the NACK fix.

@aurel32 could you coordinate with @vaussard on this?

**Comment by aurel32** [comment]
@ydamigos, thanks for the pointer. Given there is already a PR opened about that issue, I believe it's better to discuss it there. In addition the fix is now conflicting with the changes introduced by PR#5224, and it more work is needed to understand how to handle a NACK in slave mode. I'll therefore just drop that commit from the pull request.

**Comment by aurel32** [comment]
And I have just added an additional cleanup commit to make @ulfalizer happy.

**Comment by aurel32** [comment]
@ydamigos, there was a third PR with a fix for this issue, PR #5224. Given it has just been merged, i confirm that just removing the commit from my PR was the right thing to do.

**Review by ydamigos — APPROVED** [review]

File: boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig [hunk 1]
```diff
@@ -12,6 +12,9 @@ CONFIG_SERIAL=y
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 
+# enable I2C
+CONFIG_I2C=y
```
Comment (erwango) [inline boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig]: Should be 'n' by default.

**Review by erwango — CHANGES_REQUESTED** [review]
One comment.

Otherwise, could you update help message for I2C_STM32_V2. Message is not quite clear but we should be explicit that drivers now supports F7 (while previously it was 'compatible': I agree this is subtle :-))

File: boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig (hunk 1 above)
Comment (ydamigos) [reply boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig]: @erwango We enable buses on board level when there is a peripheral (e.g. WM8994 or touch panel in this case) that needs them, aren't we?

Comment (erwango) [reply boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig]: @ydamigos : we should configure them, and this is what is done here. But activation should always be done in application and not default.
Aim is that several developers on the same board know what is the base components for the board so they only enable what they need. But they don't have to bother disabling what other developers may have enabled meanwhile.

Comment (ydamigos) [reply boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig]: @erwango We need to clean up board _defconfig files then.

Comment (erwango) [reply boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig]: @ydamigos : yes, there is some work ahead! I raised https://github.com/zephyrproject-rtos/zephyr/issues/7151 earlier, waiting https://github.com/zephyrproject-rtos/zephyr/pull/7176 to be reviewed and used a reference (which was locked waiting for new SDK version and which I hope should get some feedbacks soon)

**Comment by aurel32** [comment]
@erwango: I have just done the requested changes, i.e. I updated the `Kconfig.stm32` help message to mark F7 as supported, and removed `CONFIG_I2C=y` from the board defconfig.

**Comment by aurel32** [comment]
@erwango, I have just pushed a new version to solve the conflict introduced by PR #8663 and to add the new clock properties introduced in PR #8759.

**Review by dbkinder — APPROVED** [review]
Doc changes LGTM, thanks.

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Comment by aurel32** [comment]
I have just pushed a new version to solve the conflict due the merge of USB OTG FS support. There is no code change beside solving conflicts.

**Comment by aurel32** [comment]
I have just pushed a new version to solve the conflict due the merge of USB OTG HS support. There is no code change beside solving conflicts.

**Review by nashif — APPROVED** [review]

**Pull request merged**